type-check/findIndex.test.ts

```diff
@@ -0,0 +1,26 @@
+import * as Test from "../src/types/Test";
+import { findIndex, pipe, toAsync } from "../src";
+
+const { checks, check } = Test;
+
+const res1 = findIndex((a) => a === 3, []);
+const res2 = findIndex((a) => a === 3, [1, 2, 3, 4, 5]);
+const res3 = findIndex((a) => a === "e", ["a", "b", "c", "d"]);
+
+const res4 = pipe(
+  [1, 2, 3, 4, 5],
+  findIndex((a) => a === 3),
+);
+const res5 = pipe(
+  ["a", "b", "c", "d"],
+  toAsync,
+  findIndex((a) => Promise.resolve(a === "e")),
+);
+
+checks([
+  check<typeof res1, number, Test.Pass>(),
```

Comment:
How about making it inferred to be a literal type of -1 ?

Reply:
good point 👍